pkg/operator/controller/externaldns/status_test.go

```diff
@@ -0,0 +1,668 @@
+package externaldnscontroller
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
```

Comment:
I didn't see much of added value of this package. It's used to compare strings and lengths of slices and doing the error logging.

I would avoid adding another test package and would suggest to fallback to the standard comparisons and `testing.T.Errorf()`. Because `gomega` and `ginkgo` are coming with Arjun's webhook PR which will increase the number of the testing approaches to 3: go native, testify, gingko.

I think that `ginkgo` will dominate this repository at some moment but go native way is also a possible option in some cases.